core/src/main/scala/kafka/coordinator/ConsumerMetadata.scala

```diff
@@ -46,4 +46,5 @@ private[coordinator] class ConsumerMetadata(val consumerId: String,
   var awaitingRebalanceCallback: (Set[TopicAndPartition], String, Int, Short) => Unit = null
   var assignedTopicPartitions = Set.empty[TopicAndPartition]
   var latestHeartbeat: Long = -1
+  var isLeaving: Boolean = false
```

Comment:
I want to remove any pending DelayedHeartbeat related to the leaving consumer. Purgatory's only API for this is checkAndComplete, so I needed some way to get checkAndComplete to flush out any pending DelayedHeartbeat for that consumer. checkAndComplete uses DelayedOperation.tryComplete, so I needed some way for DelayedHeartbeat.tryComplete to know that the DelayedHeartbeat is complete. I think this requires some sort of flag on the ConsumerMetadata.